components/external_server/tests/routes/test_metric.py

```diff
@@ -114,11 +115,15 @@ def test_post_metric_name(self, request):
         )
 
     def test_post_metric_type(self, request):
-        """Test that the metric type can be changed."""
+        """Test that the metric type can be changed and that sources that support the new type are not removed."""
```

Comment:
Should there not be a test that verifies changing `old_type` of `source_type2` will yield `new_type` without sources?

Reply:
The metric has two sources, one of which supports the new metric type, the other doesn't. So when the metric type is changed from old type to new type, the source that supports the new type is retained and the other is removed. I think that covers all scenario's.